### PR TITLE
Do not reconfigure for cloud backend

### PR DESCRIPTION
**Pull request by dex4er**
Cloud backend doesn't like `-reconfigure` argument and there is proper condition in the code but somehow it is hardcoded in array initialization.

### CODE REVIEW

**Comment by minamijoyo** [comment]
Hi @dex4er, Have you tried the `is_backend_terraform_cloud` option introduced in v0.3.2+? It allows you to skip the `-reconfigure` flag when switching remote state backend to the Terraform Cloud. If it doesn't help you, please open a new bug report with a reproduction case.
https://github.com/minamijoyo/tfmigrate/tree/v0.3.4#is_backend_terraform_cloud

**Comment by dex4er** [comment]
@minamijoyo of course I tried `is_backend_terraform_cloud` and it can't work correctly because it is ignored then `tfmigrate` dies because of `terraform init -reconfigure` additional option.

See: this condition https://github.com/minamijoyo/tfmigrate/blob/master/tfexec/terraform.go#L262 adds *2nd* `-reconfigure` to the array defined in https://github.com/minamijoyo/tfmigrate/blob/master/tfexec/terraform.go#L257

In other words:

Without this patch, `tfmigrate` *always* breaks with 

```
Error: Invalid command-line option

The -reconfigure option is for in-place reconfiguration of state backends
only, and is not needed when changing Terraform Cloud settings.
```

With my patch `tfmigrate` breaks with this error *only* when `is_backend_terraform_cloud = false` and works *correctly* when `is_backend_terraform_cloud = true`.

**Comment by minamijoyo** [comment]
@dex4er Thank you for your patch! It looks good but acceptance tests are failing due to an unrelated timeout issue. I tried to rerun but still failing.
I fixed the timeout issue in #100. Could you rebase this branch on the latest master?

**Comment by dex4er** [comment]
@minamijoyo please redo tests

**Review by minamijoyo — APPROVED** [review]
Thank you for the fix 👍
